server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/VacationSmtpForwardIntegrationTest.java

```diff
@@ -0,0 +1,193 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap;
+
+import static com.jayway.restassured.RestAssured.given;
+import static com.jayway.restassured.config.EncoderConfig.encoderConfig;
+import static com.jayway.restassured.config.RestAssuredConfig.newConfig;
+import static org.hamcrest.Matchers.equalTo;
+
+import java.net.InetAddress;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.collect.ImmutableList;
+import com.jayway.restassured.specification.RequestSpecification;
+import org.apache.commons.net.smtp.SMTPClient;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.james.GuiceJamesServer;
+import org.apache.james.dnsservice.api.InMemoryDNSService;
+import org.apache.james.jmap.api.access.AccessToken;
+import org.apache.james.mailbox.model.MailboxConstants;
+import org.apache.james.mailbox.store.probe.MailboxProbe;
+import org.apache.james.modules.MailboxProbeImpl;
+import org.apache.james.probe.DataProbe;
+import org.apache.james.util.streams.SwarmGenericContainer;
+import org.apache.james.utils.DataProbeImpl;
+import org.apache.james.utils.JmapGuiceProbe;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.google.common.base.Charsets;
+import com.jayway.awaitility.Awaitility;
+import com.jayway.awaitility.Duration;
+import com.jayway.awaitility.core.ConditionFactory;
+import com.jayway.restassured.builder.RequestSpecBuilder;
+import com.jayway.restassured.http.ContentType;
+import org.testcontainers.shaded.com.google.common.net.InetAddresses;
+
+public abstract class VacationSmtpForwardIntegrationTest {
+
+
+    @Rule
+    public final SwarmGenericContainer fakeSmtp = new SwarmGenericContainer("weave/rest-smtp-sink:latest");
+
+    private static final String DOMAIN = "mydomain.tld";
+    private static final String USER = "benwa";
+    private static final String USER_WITH_DOMAIN = USER + '@' + DOMAIN;
+    private static final String PASSWORD = "secret";
+    private static final String REASON = "Message explaining my wonderful vacations";
+
+    private static final String LOCALHOST_IP = "127.0.0.1";
+    private static final int SMTP_PORT = 1025;
+
+    private ConditionFactory calmlyAwait;
+    private GuiceJamesServer guiceJamesServer;
+    private JmapGuiceProbe jmapGuiceProbe;
+
+    private RequestSpecification jmapRequestSpecification;
+    private RequestSpecification restSmtpSinkSpec;
+
+    protected abstract void await();
+
+    protected abstract GuiceJamesServer getJmapServer();
+
+    protected abstract InMemoryDNSService getInMemoryDns();
+
+    @Before
+    public void setUp() throws Exception {
+
+        InetAddress containerIp = InetAddresses.forString(fakeSmtp.getIp());
+        getInMemoryDns()
```

Comment:
you should either change the InMemoryDns api to take decent parameters or extract that call in a meaningful method